__tests__/wallet-post.spec.js

```diff
@@ -0,0 +1,91 @@
+require('dotenv').config();
+const request = require('supertest');
+const {expect} = require('chai');
+const sinon = require('sinon');
+const chai = require('chai');
+const server = require('../server/app');
+const seed = require('./seed');
+chai.use(require('chai-uuid'));
+const WalletService = require('../server/services/WalletService');
+
+const {apiKey} = seed;
+
+describe('Wallet: create(POST) wallets of an account', () => {
+    let bearerTokenA;
+
+    before(async () => {
+        await seed.clear();
+        await seed.seed();
+
+        {
+            // Authorizes before each of the follow tests
+            const res = await request(server)
+                .post('/auth')
+                .set('treetracker-api-key', apiKey)
+                .send({
+                    wallet: seed.wallet.name,
+                    password: seed.wallet.password,
+                });
+
+            expect(res).to.have.property('statusCode', 200);
+            bearerTokenA = res.body.token;
+            expect(bearerTokenA).to.match(/\S+/);
+        }
+    });
+
+
+    beforeEach(async () => {
+        sinon.restore();
+    });
+
+    it('create wallet by a valid wallet name', async () => {
+        const walletService = new WalletService();
+        const res = await request(server)
+            .post('/wallets')
+            .send({wallet: 'azAZ.-@0123456789'})
+            .set('treetracker-api-key', apiKey)
+            .set('content-type', 'application/json')
+            .set('Authorization', `Bearer ${bearerTokenA}`);
+
+        expect(res.body).contain({wallet: 'azAZ.-@0123456789'})
+        expect(res.body.id).to.exist;
+        await walletService.getById(res.body.id).then((wallet) => {
+            expect(wallet.name).to.eq('azAZ.-@0123456789');
+        })
+    })
+
+    it('create wallet by invalid name length', async () => {
+        const res = await request(server)
+            .post('/wallets')
+            .send({wallet: 'ab'})
+            .set('treetracker-api-key', apiKey)
+            .set('content-type', 'application/json')
+            .set('Authorization', `Bearer ${bearerTokenA}`);
+
+        expect(res).property('statusCode').to.eq(422);
+
+        const resB = await request(server)
+            .post('/wallets')
+            .send({wallet: '2023.7.26CodingAtCanadaNiceToMeetYouuuuuuuuuuuuuuuu'})
+            .set('treetracker-api-key', apiKey)
+            .set('content-type', 'application/json')
+            .set('Authorization', `Bearer ${bearerTokenA}`);
+
+        expect(resB).property('statusCode').to.eq(422);
+    })
```

Comment:
If we already cover this edge case in unit test, then I think we don't need to do this in int test, I feel unit test is a better choice to cover these edge case:
1. It is less expensive, 
2. It is easier to build the test case.